services/user/src/user/create.js

```
@@ -19,22 +20,21 @@ async function create(event, context) {
     email: event.request.userAttributes.email,
     phoneNumber: event.request.userAttributes.phoneNumber,
     userName: event.userName,
-    createdAt: new Date().getTime(),
-    role: 'user',
+    role: 'user', // currently restricted to just 'user'
     notifications: [],
     processes: [],
     devices: [],
   }
 
   validateCreate(newUserData)
-  const db = await mongo.getDB()
-  const user = new User(db)
 
-  const result = await user.create(newUserData)
+  await connectAllModels()
+
+  const newUser = await User.create(newUserData)
 
   context.succeed(event) // triggers lambda to close via cognito request method
 
-  return success(result)
+  return success(newUser) // this return should not be doing anything if context.succeed is working as expected
```

If context fails or errors will this still return as usual? I had a quick look around for http-response in my IDE and the package.json of this repo and couldn't see it, does it have a failed method? would a flow like this possible:

```
try{
  context.succeed(event)
  return success(newUser)
}catch(e){
  return failed(e)
}
```

---

We originally had this flow you describe here in all functions. We decided: If it fails, let it fail. Otherwise you can use `context.fail` in teh catch block.

---

this handler needs some love and investigation anyway

---

I'm unsure. This was really me kicking the can down the road.

I don't believe that return can ever be hit (`context.succeed` ends execution as I understand), and on top of that the response isn't necessary because the caller doesn't make use of it. I just removed that line all together.

---

didn't see viktor's comment before I wrote
like he says. I removed the misleading return for now